openflow13/nx_action.go

```diff
@@ -0,0 +1,1000 @@
+	n += len(b)
+	binary.BigEndian.PutUint16(data[n:], a.InPort)
+	n += 2
+	data[n] = a.TableID
+	n += 1
+	// Skip padding copy, move the index.
+	n += 3
+
+	return
+}
+
+func (a *NXActionResubmitTable) UnmarshalBinary(data []byte) error {
+	n := 0
+	a.NXActionHeader = new(NXActionHeader)
+	err := a.NXActionHeader.UnmarshalBinary(data[n:])
+	n += int(a.NXActionHeader.Len())
+	if len(data) < int(a.Len()) {
+		return errors.New("the []byte is too short to unmarshal a full NXActionResubmitTable message")
+	}
+	a.InPort = binary.BigEndian.Uint16(data[n:])
+	n += 2
+	a.TableID = data[n]
+	n += 1
+	// Skip padding copy, move the index.
+	n += 3
+
+	return err
+}
+
+func NewNXActionResubmitTableCT(inPort uint16, tableId uint8) *NXActionResubmitTable {
+	a := newNXActionResubmitTableCT()
+	a.InPort = inPort
+	a.TableID = tableId
+	return a
+}
+
+func NewNXActionResubmitTableCTNoInPort(tableId uint8) *NXActionResubmitTable {
+	a := newNXActionResubmitTableCT()
+	a.InPort = OFPP_IN_PORT
+	a.TableID = tableId
+	return a
+}
+
+// nxast_resubmit_table
+type NXActionCTNAT struct {
+	*NXActionHeader
+	pad          []byte // 2 bytes
+	Flags        uint16 // nat Flags to identify snat/dnat, and nat algorithms: random/protocolHash, connection persistent
+	rangePresent uint16 // mark if has set nat range, including ipv4/ipv6 range, port range
+
+	rangeIPv4Min  net.IP
+	rangeIPv4Max  net.IP
+	rangeIPv6Min  net.IP
+	rangeIPv6Max  net.IP
+	rangeProtoMin *uint16
+	rangeProtoMax *uint16
+}
+
+func NewNXActionCTNAT() *NXActionCTNAT {
+	a := new(NXActionCTNAT)
+	a.NXActionHeader = NewNxActionHeader(NXAST_NAT)
+	a.Length = 16
+	a.pad = make([]byte, 2)
+	return a
+}
+
+func (a *NXActionCTNAT) Len() (n uint16) {
+	a.Length = ((a.Length + 7) / 8) * 8
+	return a.Length
+}
+
+func (a *NXActionCTNAT) MarshalBinary() (data []byte, err error) {
+	data = make([]byte, a.Len())
+	b := make([]byte, a.NXActionHeader.Len())
+	n := 0
+
+	b, err = a.NXActionHeader.MarshalBinary()
+	copy(data[n:], b)
+	n += len(b)
+	// Skip padding bytes
+	n += 2
+	binary.BigEndian.PutUint16(data[n:], a.Flags)
+	n += 2
+	binary.BigEndian.PutUint16(data[n:], a.rangePresent)
+	n += 2
+
+	if a.rangeIPv4Min != nil {
+		copy(data[n:], a.rangeIPv4Min.To4())
+		n += 4
+	}
+	if a.rangeIPv4Max != nil {
+		copy(data[n:], a.rangeIPv4Max.To4())
+		n += 4
+	}
+	if a.rangeIPv6Min != nil {
+		copy(data[n:], a.rangeIPv6Min.To16())
+		n += 16
+	}
+	if a.rangeIPv6Max != nil {
+		copy(data[n:], a.rangeIPv6Max.To16())
+		n += 16
+	}
+	if a.rangeProtoMin != nil {
+		binary.BigEndian.PutUint16(data[n:], *a.rangeProtoMin)
+		n += 2
+	}
+	if a.rangeProtoMin != nil {
+		binary.BigEndian.PutUint16(data[n:], *a.rangeProtoMax)
+		n += 2
+	}
+
+	return
+}
+
+func (a *NXActionCTNAT) SetSNAT() error {
+	if a.Flags&NX_NAT_F_DST != 0 {
+		return errors.New("the SNAT and DNAT actions should be mutually exclusive")
+	}
+	a.Flags |= NX_NAT_F_SRC
+	return nil
+}
+
+func (a *NXActionCTNAT) SetDNAT() error {
+	if a.Flags&NX_NAT_F_SRC != 0 {
```

Comment:
Please run gofmt on the file.

Reply:
Hi @jojimt , I have run command "gofmt -l -d -w openflow13/*.go" to format the code, but it seems no errors obey the check. Do I miss some parmateters or configuraitons in the command? Could you help give a correct command or configuration example?

Reply:
gofmt -s -w is enough.

Reply:
Actually, even if I add spceses around & manually, after running gofmt command, the spaces are removed. So I don't think it is a format issue.

Reply:
What version of go are you using?

Reply:
go version go1.12.7 linux/amd64